game/game.go

```diff
@@ -199,6 +199,7 @@ func (g *Game) TakeAction(action *Action) {
 	}
 
 	if action.Type == UseForMana {
+		action.With.Owner = g.Priority()
```

Comment:
this doesn't seem right, why would a mana action change the owner of a card? the owner of a card should never change... maybe you control magic'd a llanowar elf.